src/Ocelot/Configuration/Creator/UpstreamTemplatePatternCreator.cs

```diff
@@ -40,7 +40,15 @@ public UpstreamPathTemplate Create(IRoute route)
         if (upstreamTemplate.Contains('?'))
         {
             containsQueryString = true;
-            upstreamTemplate = upstreamTemplate.Replace("?", "(|\\?)");
+
+            if (upstreamTemplate.Contains("/?"))
+            {
+                upstreamTemplate = upstreamTemplate.Replace("/?", "(\\/|\\?|$)");
+            }
+            else if (upstreamTemplate.Contains('?'))
+            {
+                upstreamTemplate = upstreamTemplate.Replace("?", "(\\/|\\?|$)");
+            }

```

Comment:
The `else` branch is unnecessary as the condition is already evaluated on line 40.
Why not combine both branches of the `if` statement into a single coalescing operator?
```suggestion
            var ampersand = upstreamTemplate.Contains("/?")) ? "/?" : "?";
            upstreamTemplate = upstreamTemplate.Replace(ampersand, "(\\/|\\?|$)");
```

Reply:
Don't you like this suggestion?

Reply:
You are definitely right about the unnecessary `else if ` statement and I removed it. :D

Reply:
Gui also requested ternary operator and I got overruled :D Changed it